### PR TITLE
open decks and replays from file system

### PR DESCRIPTION
**Pull request by djpiper28**
uses qlocalserer and qlocalpeer with filelocks to make clicked links in file system open in the active cockatrice instance. requires the mime type to be installed to work. linux only at the moment as windows requires registry changes to get it to be recognised as the default application for .cor/.cod files. all platforms should work but would need the user to manually select it.

## Related Ticket(s)
none

## Short roundup of the initial problem
Replays and decks must be opened in the client, this adds a feature to open them from the file system.

## What will change with this Pull Request?
- added qtlocalpeer which is the backend to the inter process communication
- added the code to parse the arguments that are parsed by xdg-open (separate from the current argument parser)
- the application instance manager is a short wrapper class from qbittorrent which makes the qtlocalpeer easier to work with
- added the mime type definition and the script to install the mimes on a linux system

## Screenshots

https://youtu.be/LMScvDbK4a8 is a video of it working

### CODE REVIEW

File: cockatrice/src/dlg_connect.cpp [hunk 1]
```diff
@@ -332,6 +332,36 @@ QString DlgConnect::getHost() const
     return hostEdit->text().trimmed();
 }
 
+void DlgConnect::setServer(QString serverName, QString address, unsigned int port)
```
Comment (djpiper28) [inline cockatrice/src/dlg_connect.cpp:335]: this code should not be here my bad

**Comment by ebbit1q** [comment]
:eyes: looks like it actually compiled

**Comment by djpiper28** [comment]
It does compile but I don't like using th home folders name for the instance id. But it does compile so I might not change it.

**Comment by djpiper28** [comment]
> How's this coming along? Is it ready to be tested?

Yeah it works, however the file association has to be setup manually.

**Pull request closed without merging**